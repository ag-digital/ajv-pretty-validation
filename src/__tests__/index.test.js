import Ajv from 'ajv'
import prettyValidation from '../index'

describe('prettyValidation()', () => {
  const ajv = new Ajv({
    allErrors: true,
    extendRefs: 'fail'
  })

  ajv.addSchema(
    {
      additionalProperties: false,
      type: 'object',
      properties: {
        name: {
          type: 'string'
        },
        surname: {
          type: 'string'
        },
        address: {
          additionalProperties: false,
          type: 'object',
          properties: {
            line1: { type: 'string' },
            city: { type: 'string' }
          },
          required: ['line1', 'city']
        },
        contactInfo: {
          additionalProperties: false,
          type: 'object',
          properties: {
            telephone: { type: 'string' },
            email: { type: 'string' }
          },
          required: ['telephone']
        },
        favouriteFruit: {
          additionalItems: false,
          items: {
            type: 'string'
          },
          type: 'array',
          uniqueItems: true
        },
        lovesJson: {
          type: 'boolean'
        }
      },
      required: ['name', 'surname', 'address', 'contactInfo']
    },
    'person'
  )

  const data = {
    name: 'Sean',
    address: {
      line1: 1,
      city: 'london'
    },
    contactInfo: {
      email: 'foo@bar.com'
    },
    favouriteFruit: [
      'banana',
      'banana'
    ],
    lovesJson: 'true'
  }

  it('prints the expected validation output', () => {
    ajv.validate('person', data)
    const actual = prettyValidation(ajv.errors, data)
    console.log(actual)
    expect(actual).toMatchSnapshot()
  })
})
