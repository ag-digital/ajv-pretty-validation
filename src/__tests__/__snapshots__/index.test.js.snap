// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prettyValidation() prints the expected validation output 1`] = `
"{
  \\"name\\": \\"Sean\\"
  \\"address\\": {
    [31m\\"line1\\": 1[39m
    [35m^ \\"line1\\" should be string[39m
    \\"city\\": \\"london\\"
  }
  [31m\\"contactInfo\\": {[39m
[31m    \\"email\\": \\"foo@bar.com\\"[39m
[31m  }[39m
  [35m^ \\"contactInfo\\" should have required property 'telephone'[39m
  [31m\\"favouriteFruit\\": [\\"banana\\", \\"banana\\"][39m
  [35m^ \\"favouriteFruit\\" should NOT have duplicate items (items ## 0 and 1 are identical)[39m
  [31m\\"lovesJson\\": \\"true\\"[39m
  [35m^ \\"lovesJson\\" should be boolean[39m
}"
`;
